api/osf_groups/views.py

```diff
@@ -0,0 +1,131 @@
+    model_class = apps.get_model('osf.OSFGroup')
+
+    parser_classes = (JSONAPIMultipleRelationshipsParser, JSONAPIMultipleRelationshipsParserForRegularJSON,)
+    serializer_class = OSFGroupSerializer
+    view_category = 'osf_groups'
+    view_name = 'group-list'
+    ordering = ('-modified', )
+
+    def get_default_queryset(self):
+        return OSFGroup.objects.all()
+
+    # overrides ListCreateAPIView
+    def get_queryset(self):
+        return self.get_queryset_from_request()
+
+    # overrides ListCreateAPIView
+    def perform_create(self, serializer):
+        """Create an OSFGroup.
+
+        :param serializer:
+        """
+        # On creation, logged in user is the creator
+        user = self.request.user
+        serializer.save(creator=user)
+
+
+class OSFGroupDetail(JSONAPIBaseView, generics.RetrieveUpdateDestroyAPIView, OSFGroupMixin):
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+        IsGroupManager,
+    )
+    required_read_scopes = [CoreScopes.OSF_GROUPS_READ]
+    required_write_scopes = [CoreScopes.OSF_GROUPS_WRITE]
+    model_class = apps.get_model('osf.OSFGroup')
+
+    parser_classes = (JSONAPIMultipleRelationshipsParser, JSONAPIMultipleRelationshipsParserForRegularJSON,)
+    serializer_class = OSFGroupDetailSerializer
+    view_category = 'osf_groups'
+    view_name = 'group-detail'
+    ordering = ('-modified', )
+
+    # Overrides RetrieveUpdateDestroyAPIView
+    def get_object(self):
+        return self.get_osf_group()
+
+    # Overrides RetrieveUpdateDestroyAPIView
+    def perform_destroy(self, instance):
+        auth = get_user_auth(self.request)
+        instance.remove_group(auth=auth)
+
+
+class OSFGroupManagersList(JSONAPIBaseView, generics.ListAPIView, OSFGroupMixin):
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+    )
+    required_read_scopes = [CoreScopes.OSF_GROUPS_READ]
+    required_write_scopes = [CoreScopes.NULL]
+
+    model_class = apps.get_model('osf.OSFUser')
+    serializer_class = UserSerializer
+    view_category = 'osf_groups'
+    view_name = 'group-managers'
+    ordering = ('-modified', )
+
+    # Overrides ListAPIView
+    def get_queryset(self):
+        return self.get_osf_group().managers
+
+
+class OSFGroupMembersList(JSONAPIBaseView, generics.ListAPIView, OSFGroupMixin):
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+    )
+    required_read_scopes = [CoreScopes.OSF_GROUPS_READ]
+    required_write_scopes = [CoreScopes.NULL]
+
+    model_class = apps.get_model('osf.OSFUser')
+    serializer_class = UserSerializer
+    view_category = 'osf_groups'
+    view_name = 'group-members'
+    ordering = ('-modified', )
+
+    # Overrides ListAPIView
+    def get_queryset(self):
+        return self.get_osf_group().members_only
```

Comment:
Managers belong to an OSF Group's manager django group, but all members (members AND managers) belong to the member django group.  OSFGroup.managers returns managers only OSFGroup,.members returns members/managers, OSFGropu.members_only returns members only.